Tonuino.ino

```diff
@@ -404,35 +407,37 @@ class KindergardenMode: public Modifier {
 
   public:
     virtual bool handleNext() {
-      Serial.println(F("== KindergardenMode::handleNext() -> NEXT"));
-      //if (this->nextCard.cookie == cardCookie && this->nextCard.nfcFolderSettings.folder != 0 && this->nextCard.nfcFolderSettings.mode != 0) {
-      //myFolder = &this->nextCard.nfcFolderSettings;
+      DEBUG_PRINTLN(F("== KindergardenMode::handleNext() -> NEXT"));
       if (this->cardQueued == true) {
         this->cardQueued = false;
 
         myCard = nextCard;
         myFolder = &myCard.nfcFolderSettings;
-        Serial.println(myFolder->folder);
-        Serial.println(myFolder->mode);
+        DEBUG_PRINTLN(myFolder->folder);
+        DEBUG_PRINTLN(myFolder->mode);
         playFolder();
         return true;
       }
       return false;
     }
-    //    virtual bool handlePause()     {
-    //      Serial.println(F("== KindergardenMode::handlePause() -> LOCKED!"));
-    //      return true;
-    //    }
-    virtual bool handleNextButton()       {
-      Serial.println(F("== KindergardenMode::handleNextButton() -> LOCKED!"));
+    virtual bool handleNextButton() {
+      if (currentTrack != numTracksInFolder) {
+        currentTrack = currentTrack + 1;
+        mp3.playFolderTrack(myFolder->folder, currentTrack);
+        return false;
+      }
```

Comment:
Did you change this on purpose? As I understand it, the idea behind "KindergardenMode" is that whatever the kids do, they cannot change a track while it's playing. They can only queue new tracks to be played after the current one has finished. That's why the nextButton and previousButton are locked in this mode. This is probably to prevent Wikipedia-style "edit wars" in Kindergarden. :)

I have never used KindergardenMode myself, but I believe the change to this button function should probably be reverted. Same for handlePreviousButton below.

Reply:
In the context I've committed the change, it was an accident. But really, I don't like the Modifiers. I'll remove them in by Repo soon. The way Modifiers are implemented is the right approach - for Modes. But there is no reason for two mechanisms which basically do the same.